packages/web-components/src/styles/partials/badge.partials.ts

```diff
@@ -0,0 +1,299 @@
+import { css } from '@microsoft/fast-element';
+import {
+  colorBrandBackground,
+  colorBrandBackground2,
+  colorBrandForeground1,
+  colorBrandForeground2,
+  colorBrandStroke2,
+  colorNeutralBackground1,
+  colorNeutralBackground4,
+  colorNeutralBackground5,
+  colorNeutralForeground1,
+  colorNeutralForeground3,
+  colorNeutralForegroundInverted,
+  colorNeutralForegroundOnBrand,
+  colorNeutralForegroundStaticInverted,
+  colorNeutralStroke2,
+  colorNeutralStrokeAccessible,
+  colorPaletteDarkOrangeBackground1,
+  colorPaletteDarkOrangeBackground3,
+  colorPaletteDarkOrangeBorder1,
+  colorPaletteDarkOrangeForeground1,
+  colorPaletteDarkOrangeForeground3,
+  colorPaletteGreenBackground1,
+  colorPaletteGreenBackground3,
+  colorPaletteGreenBorder2,
+  colorPaletteGreenForeground1,
+  colorPaletteGreenForeground2,
+  colorPaletteGreenForeground3,
+  colorPaletteRedBackground1,
+  colorPaletteRedBackground3,
+  colorPaletteRedBorder1,
+  colorPaletteRedForeground1,
+  colorPaletteRedForeground3,
+  colorPaletteYellowBackground1,
+  colorPaletteYellowBackground3,
+  colorPaletteYellowBorder1,
+  colorPaletteYellowForeground2,
+  colorTransparentStroke,
+  fontSizeBase100,
+  fontSizeBase200,
+  lineHeightBase100,
+  lineHeightBase200,
+  spacingHorizontalSNudge,
+  spacingHorizontalXS,
+  spacingHorizontalXXS,
+} from '../../theme/design-tokens.js';
+
+const textPadding = spacingHorizontalXXS;
+
+/**
+ * @public
+ * The badge's size styles
+ */
+export const badgeSizeStyles = css.partial`
```

Comment:
What is the motivation for having the partials away from the component? I would expect these to be in `badge`, owned by it and other component(s) importing from there.

Reply:
I was primarily thinking of keeping shared things in a shared location which has greater visibility than within the component styles file as that could get bulky. Additionally one thought that I had was that not all partials may be as intuitive with regard to what the actual "primary" component this would be inheriting from. Consider scenarios where we have the same visuals across a functional button/anchor for example or across text field/number field, or select and combobox (currently two implementations because the behavior and functionality is different, we'll need to dig in here I know). In those cases, which one does the styles live in? They're kind of both on the same level with shared appearances but where would I look for the shared styles? In my mind having a dedicated place where we keep composable styles or partials made a lot of sense.

Reply:
Ok, let's start with a shared location and see how that goes (and tree-shakes).
From my experience, when an engineer will need to update the shared style, they will be afraid to change it and will rather copy it anyway. But I strongly believe we will do better :-D

Reply:
Let's strive to do just that!